UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common.UnitTests/Helpers/FileShareServiceCacheTest.cs

```diff
@@ -0,0 +1,174 @@
+                                                           new Attribute { Key= "CellName", Value= "DE416050" },
+                                                           new Attribute { Key= "EditionNumber", Value= "2" } ,
+                                                           new Attribute { Key= "UpdateNumber", Value= "0" },
+                                                           new Attribute { Key= "ProductCode", Value= "AVCS" }}
+            };
+        }
+
+        private FssResponseCache GetResponseCache()
+        {
+            return new FssResponseCache()
+            {
+                BatchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272",
+                PartitionKey = "DE416050",
+                RowKey = "2|0",
+                Response = JsonConvert.SerializeObject(GetBatchDetail())
+            };
+        }
+        [Test]
+        public async Task WhenGetNonCacheProductDataForFssReturnsProductListFound()
+        {
+            string exchangeSetRootPath = @"C:\\HOME";          
+
+            A.CallTo(() => fakeAzureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(GetStorageAccountConnectionStringAndContainerName().Item1);
+            A.CallTo(() => fakeAzureTableStorageClient.RetrieveAsync<FssResponseCache>(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(GetResponseCache());
+            A.CallTo(() => fakeAzureBlobStorageClient.GetCloudBlockBlob(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(new CloudBlockBlob(new System.Uri("http://tempuri.org/blob")));
+            A.CallTo(() => fakeFileSystemHelper.DownloadToFileAsync(A<CloudBlockBlob>.Ignored, A<string>.Ignored));
+
+            var response = await fileShareServiceCache.GetNonCacheProductDataForFss(GetProductdetails(), GetSearchBatchResponse(), exchangeSetRootPath, GetScsResponseQueueMessage(), null, CancellationToken.None);
+
+            Assert.IsNotNull(response);
+            Assert.IsInstanceOf(typeof(List<Products>), response);
+        }
+
+        [Test]
+        public async Task WhenGetNonCacheProductDataForFssReturnsProductListNotFound()
+        {
+            var cachingResponse = new FssResponseCache() { };
+            A.CallTo(() => fakeAzureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns(GetStorageAccountConnectionStringAndContainerName().Item1);
+            A.CallTo(() => fakeAzureTableStorageClient.RetrieveAsync<FssResponseCache>(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(cachingResponse);
+
+            var response = await fileShareServiceCache.GetNonCacheProductDataForFss(GetProductdetails(), null, string.Empty, GetScsResponseQueueMessage(), null, CancellationToken.None);
+
+            Assert.IsNotNull(response);
+            Assert.IsInstanceOf(typeof(List<Products>), response);
```

Comment:
Kindly assert count in first 2 test cases
Assert.AreEqual(0, response.Count);
Assert.AreEqual(1, response.Count);

and remove Assert.IsInstanceOf(typeof(List<Products>), response); from first 2 test cases as it is not required.